### PR TITLE
Remove the project dependency of the helper-cli on the reporter

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: model/src/main/kotlin/utils/DefaultResolutionProvider.kt [hunk 1]
```diff
@@ -17,15 +17,15 @@
...
+package org.ossreviewtoolkit.model.utils
 
 import org.ossreviewtoolkit.model.OrtIssue
 import org.ossreviewtoolkit.model.OrtResult
 import org.ossreviewtoolkit.model.RuleViolation
 import org.ossreviewtoolkit.model.config.Resolutions
 
 /**
- *
+ * A default provider of resolutions for [OrtIssue]s.
```
Comment (sschuberth) [inline model/src/main/kotlin/utils/DefaultResolutionProvider.kt:28]: @mnonnenmacher I'd appreciate if you can suggesting some more meaningful docs here that explains *how* issues are resolved by default.

Comment (sschuberth) [reply model/src/main/kotlin/utils/DefaultResolutionProvider.kt:28]: I want to get this merged, will eventually improve the wording later.

File: model/src/main/kotlin/utils/ResolutionProvider.kt [hunk 2]
```diff
@@ -27,7 +27,7 @@ import org.ossreviewtoolkit.model.config.Resolutions
 import org.ossreviewtoolkit.model.config.RuleViolationResolution
 
 /**
- * A provider for resolutions of [OrtIssue]s.
+ * An interface to provide resolutions for [OrtIssue]s.
```
Comment (mnonnenmacher) [inline model/src/main/kotlin/utils/ResolutionProvider.kt:30]: This is outdated, it also provides resolutions vor `[RuleViolation]s`.

File: model/src/main/kotlin/utils/DefaultResolutionProvider.kt (hunk 1 above)
Comment (mnonnenmacher) [reply model/src/main/kotlin/utils/DefaultResolutionProvider.kt:28]: This also provides resolutions for `[RuleViolation]s`. The only thing I would add here is how this class works, something like "resolutions can be added using the [add] function". How resolutions work is documented in the respective `RuleViolationResolution` and `IssueResolution` (which should maybe be renamed to `OrtIssueResolution`) classes.

File: model/src/main/kotlin/utils/ResolutionProvider.kt (hunk 2 above)
Comment (sschuberth) [reply model/src/main/kotlin/utils/ResolutionProvider.kt:30]: Thanks, please see https://github.com/oss-review-toolkit/ort/pull/2655.

**Pull request merged**